### ISSUE TITLE
hack/append/prepend: --commit and --propose flags

### ISSUE THREAD

**Issue by kevgo** [git-town/git-town#4376]
The [boy scouts rule](https://innolution.com/resources/glossary/boy-scout-rule) is a  best practice that tooling like Git Town should make easy to perform, so that it happens more often. From https://egoless.engineering:

> We also needed to create the strong expectation that you’d leave things in a better state as you messed with other people’s stuff. That made people overjoyed to have you messing with their stuff. 

The way this is often done is to commit the cleanup into the same branch. Then either ship it as part of the branch, or extract the cleanup commit into its own branch. The former has the problem of mixing concerns, the latter is cumbersome because it requires running several commands: create the cleanup commit, create a new branch, move the commit, and propose the branch.

The most straightforward solution would be to commit the cleanup directly into a separate branch. Right now that also requires multiple commands to do. It would be nice if it was possible using a single command.

```zsh
> git town hack new-branch --commit
git checkout -b new-branch main
git commit
git checkout <old branch>
```

- `--commit` runs `git commit` on the new branch (requires stashed changes)
- `--propose` automatically proposes the new branch
- `--commit-message` commits using the given commit message (implies `--commit`)
- `--commit-all` stages all changes before committing (implies `--commit`)

**Comment by kevgo** [comment]
I would use this feature all the time. My development workflow is far more efficient when I stick to the single-responsibility principle for Git branches (or "KISS" – Keep It Simple and Straightforward). Encountering unrelated cleanups during coding is common, and my current approach is either to jot down the idea for later or create a separate branch for it immediately. Without a workflow like this, several issues arise:

1. **Context switching kills productivity:** Humans are notoriously bad at multitasking, especially in complex cognitive work like software development. Every time I switch tasks, it disrupts my flow, and it can take several minutes to regain full context and productivity.

2. **Half-implemented ideas lead to suboptimal results:** Tackling two separate changes simultaneously often results in neither being implemented well. Even "simple" changes require careful thought to identify all relevant areas to update and whether the change should address broader root causes. It's more productive to focus on one thing at a time and do it thoroughly.

3. **Multitasking issues resurface repeatedly:** These problems aren't confined to coding alone. They show up again during code review prep, where the mixed changes make it harder to validate correctness, and for reviewers, who must mentally untangle unrelated changes.

4. **Debugging becomes harder:** When unrelated changes are mixed together, tracking down the source of an issue requires extra effort to isolate which change caused them.

5. **One change takes the other one down with it:** If for some reason the branch doesn't get shipped, for example because the idea gets implemented in a different way in another branch, the unrelated cleanups are lost.

This feature would enable a much smoother workflow:

- While working on a feature, if I encounter an unrelated issue, I wouldn’t need to context-switch to address it immediately.
- Instead, I commit a simple initial fix or just a quick `// TODO: fix this` into a separate branch with a single CLI command. Okay to keep things half-finished and broken for now.
- Once the feature is complete, I can switch context to the new branch and give the unrelated fix my full attention.

**Comment by kevgo** [comment]
The command name "commit" is probably too misleading here. It isn't intuitive that "committing" also creates a whole new branch. Additionally, `git town commit --prepend` and `--append` sound like the commit is getting appended or prepended to another commit.

Proposal for better syntax:

- `git hack --commit --propose`
- `git prepend --commit --propose`
- `git append --commit --propose`

Advantages:

- makes it more clear that a new branch gets created and something gets committed into it (and the the branch is getting proposed also)
- does not require a new top-level command (Git Town has enough of them already)

**Comment by kevgo** [comment]
Open question: which branch should the `hack`/`append`/`prepend` command with the `--commit` or `--propose`  flag end up on?

Normally these commands end up on the new branch they create. Doing so in this scenario would be similar to running `git hack <branch> && git commit && git propose`
- Ending up on a branch that is already committed and proposed is kind of pointless since all the work on it is already done.
- The `--commit` and `--propose` flags wouldn't do enough to justify their existence in this scenario. 

Going back to the original branch would be similar to `git hack <branch> && git commit && git propose && git checkout -`
- This scenario serves a purpose: committing an unrelated change to another branch and keep developing on the current branch.

Proposed decision: if the user provides either the `--commit` or `--propose` flag, return to the original branch.

**Comment by kevgo** [comment]
This is now available as part of [v18.2](https://github.com/git-town/git-town/releases/tag/v18.2.0).
